### DISCUSSION TITLE
Proxmox VE Host Backup

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#1770] [Announcements]
This script serves as a versatile backup tool for specific files and directories found within the 'etc' directory. It was developed in response to numerous requests for a Proxmox host backup script. The unique feature of this script is its flexibility, it allows users to choose which files and directories to back up, making it suitable for various hosts, not just Proxmox. By default, the compressed backup file is stored in the 'root' directory.

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/host-backup.sh)"
```
~I haven't published this script on the website yet. Before making it official,~ I'd like to gather user input and feedback.

**Comment by tteck** [comment]
Backup considerations
```
aliases
hostname
hosts
lvm
network
pve
resolv.conf
sysctl.conf
vzdump.conf
```
These are speculative guesses.

**Reply by tteck** [reply]
When creating a backup of the specified files and directories in my system, the resulting tar file size is 2.6MB.

**Reply by Smile2Grow** [reply]
Thank you @tteck! Your backup consideration fits for me. Maybe a query makes sense to create exactly this selection or Free Selection as a backup?

**Comment by tteck** [comment]
Just to mention, I utilize File Browser on my Proxmox host to handle files, including tasks such as downloading and uploading.

**Reply by tteck** [reply]
If you find that File Browser doesn't meet your expectations, the uninstallation process is straightforward, just re-execute the script.

**Reply by Smile2Grow** [reply]
The filebrowser makes sense.

**Comment by tteck** [comment]
The script is currently limited to the `/etc/` directory. Would it be more beneficial to allow users to specify the directory they want to work in, such as `/root/`, `/var/`, or even `/usr/local/bin/`?

**Reply by tteck** [reply]
By implementing this approach, you could back up `/etc/pve/`, and then, if needed, rerun the script to create a backup for something located in `/root/` or any other directory of your choice.

**Reply by tteck** [reply]
Also, allow users to specify the backup path.

**Reply by tteck** [reply]
The changes have been implemented.

**Reply by tteck** [reply]
For consistency, it's necessary to include slashes both before and after the backup path and directory path when entering them.

**Comment by endigma** [comment]
What's the restore story?

**Reply by tteck** [reply]
Normal tar rules apply. The -C option specifies the destination directory where the contents of the tarball will be extracted, and if files with the same names already exist in that destination directory, they will be overwritten.

`tar -xzf backup-name.tar.gz -C /path/to/destination/directory`

e.g. `tar -xzf server-var-lib-pve-cluster-backup-2023_09_09.tar.gz -C /var/lib/pve-cluster`

**Reply by mgrimace** [reply]
What about a script that we could use after reinstalling proxmox that asks where the backup was saved, what was backed up (e.g., full, manual), and then automates the restore?

**Comment by yodaphone** [comment]
this is a good one. worked for me

can we make the selection of directories stick so that we dont have to do this everytime the script is run

**Reply by tteck** [reply]
No, my apologies.

**Comment by hvddrift** [comment]
Hi, love all the scripts, (new proxmox user) this one has me a bit stumped. 

Not sure if I have missed something entirely, but if I have a SMB attached drive  /mnt/pve/unraidpve13
I select /etc and then for example [x] PVE

Then I get this error.

Finished 
A backup is rendered ineffective when it remains stored on the host.

**Reply by tteck** [reply]
How is that an error?

**Comment by Jens-Wymeersch** [comment]
@tteck as always, this is a very useful script. Is there any way to automate it with a schedule ?
Secondly, a dropdown of all the files, creates a lot of flexibility but means that you have to click a lot. Maybe it is possible to have a suggestion built in (default). What do you think ?

**Reply by endigma** [reply]
cron

**Reply by Jens-Wymeersch** [reply]
Never did a cron job before. Probably should learn. But what about the files themselves - a subsection that is really needed instead of full autonomy ?

**Reply by tteck** [reply]
Since the Proxmox configuration is not expected to undergo frequent changes, I believe there's no necessity for a crontab job.

**Reply by Jens-Wymeersch** [reply]
Understood. What about a subsection of the files in /etc/ that is needed to be backup'ed. What is the advice there ?

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/discussions/1770#discussioncomment-6942976

**Reply by endigma** [reply]
If you want like advanced backups you can use restic or borg or etc they have docs on scheduling and including different files.
